src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/AuthorizationRepository.kt

```diff
@@ -23,6 +23,43 @@ class OrderlyAuthorizationRepository : AuthorizationRepository
     private val PERMISSION_NAME = "permission_name"
     private val GROUP_PERMISSION_ID = "permission_id"
 
+    override fun createUserGroup(userGroup: String)
+    {
+        JooqContext().use {
+
+            if (it.dsl.selectFrom(Tables.ORDERLYWEB_USER_GROUP)
+                            .where(Tables.ORDERLYWEB_USER_GROUP.ID.eq(userGroup))
+                            .singleOrNull() != null)
+            {
+                throw DuplicateKeyError(mapOf("user-group" to userGroup))
+            }
+
+            it.dsl.newRecord(Tables.ORDERLYWEB_USER_GROUP)
+                    .apply {
+                        this.id = userGroup
+                    }.store()
+        }
+    }
+
+    override fun ensureGroupHasMember(userGroup: String, email: String)
+    {
+        JooqContext().use {
+            it.dsl.selectFrom(ORDERLYWEB_USER_GROUP)
+                    .where(ORDERLYWEB_USER_GROUP.ID.eq(userGroup))
+                    .singleOrNull() ?: throw UnknownObjectError(userGroup, "user-group")
+
+            it.dsl.selectFrom(ORDERLYWEB_USER)
+                    .where(ORDERLYWEB_USER.EMAIL.eq(email))
+                    .singleOrNull() ?: throw UnknownObjectError(email, User::class)
+
+            it.dsl.newRecord(ORDERLYWEB_USER_GROUP_USER)
```

Comment:
Should this check that user isn't already a member of a group before doing the insert? 'Ensure' implies that it's fine if it is already (won't error) but we wouldn't want duplicate rows.